### ISSUE TITLE
Tinc sometimes tries to connect to other nodes via its own virtual interface

### ISSUE THREAD

**Issue by gsliepen** [gsliepen/tinc#332]
Tinc autodiscovers addresses of other nodes via its own virtual interface, and tries to use that to connect to other nodes. Find some way to avoid accepting UDP and TCP packets from tinc's own interface.

**Comment by dechamps** [comment]
Are you sure you observed this in 1.1? The 1.1 autodiscovery protocol does not use broadcasts, it advertises the locally bound address of the connection socket instead. Given this, I'm not sure how tinc 1.1 could end up sending autodiscovery packets through the tunnel.

Although on second thought, I guess someone could have set up a default route that goes through the tunnel, and that default route could indeed end up being selected for sending autodiscovery packets.

**Comment by gsliepen** [comment]
It's rare that it happens, and I'm not sure exactly how to reproduce it without explicitly forcing it like setting up a default route through the tunnel. But I've seen it move a meta-connection from the real network over to the VPN, and it even kept working correctly. There's multiple reports of this. I don't think it requires the autodiscovery protocol, but I think the likelyhood of it occuring might be higher with autodiscovery enabled.

**Comment by fangfufu** [comment]
Are those other reports Github issues? Could you post their links here please?

**Comment by fangfufu** [comment]
Putting down this as janitorial for now, as it is not clear to me what the negative impact to user experiences are.

**Comment by gsliepen** [comment]
No, those other reports were via IRC. This is not a janitorial issue, it affects performance and possibly can cause connectivity issues.

**Comment by fangfufu** [comment]
Okay, I guess we should label it with bug then.
